lib/action.js

```diff
@@ -222,29 +222,53 @@ function action(system) {
 	});
 
 	self.system.on('action_running_get', function (page, bank, cb) {
-		cb(self.actions_running.indexOf(page + '_' + bank) !== -1);
+		cb(self.actions_running.has(`${page}_${bank}`));
 	});
 
 	// If a user wants to panic-abort all timers running
 	self.system.on('action_delayed_abort', function() {
 
 		debug("Aborting delayed actions");
 
-		while(self.timers_running.length > 0) {
+		for (let timer of self.timers_running.keys()) {
 			debug("clearing timer");
-			clearTimeout( self.timers_running.shift() );
+			clearTimeout(timer);
 		}
+		self.timers_running.clear()
 
-		var actions_running = self.actions_running.slice(0); //clone hack
-		self.actions_running = []; // clear the array
+		var actions_running = self.actions_running; //clone hack
+		self.actions_running = new Set(); // clear the array
 
-		for (var bid in actions_running) {
-			var a = actions_running[bid].split("_");
+		for (let bid of actions_running.keys()) {
+			const a = bid.split("_");
 			self.system.emit('graphics_bank_invalidate', a[0], a[1]);
 		}
 
 	});
 
+	// If a user wants to abort a single button actions
+	self.system.on('action_abort_bank', function(page, bank) {
+
+		var bid = page + '_' + bank;
+		var cleared = 0;
+
+		self.actions_running.delete(bid);
+
+		self.timers_running.forEach((timerId, timer) => {
```

Comment:
oops, I forgot to delete the timers from self.timers_running. This block of code is only calling clearTimeout on them, so will be leaking the refernces...

Reply:
Fixed.

Reply:
Perfect :)